js/apps/admin-ui/src/realm-settings/OpenIdEndpointConigurationTab.tsx

```diff
@@ -0,0 +1,175 @@
+                : [];
+              return (
+                <>
+                  {claimsSupported.map((claim: string, index: number) => {
+                    if (claim) {
+                      return (
+                        <InputGroup key={index}>
+                          <TextInput
+                            id={"textInput" + index}
+                            value={claim}
+                            onChange={(value) => {
+                              claimsSupported[index] = value;
+                              field.value &&
+                                (field.value.claimsSupported =
+                                  claimsSupported.join(","));
+                              field.onChange(field.value);
+                            }}
+                            aria-label="input with button"
+                          />
+                          <Button
+                            id={"inputButtonRemove" + index}
+                            icon={<MinusIcon />}
+                            onClick={() => {
+                              claimsSupported.splice(index, 1);
+                              field.value &&
+                                (field.value.claimsSupported =
+                                  claimsSupported.join(","));
+                              field.onChange(field.value);
+                            }}
+                            variant="control"
+                          />
+                        </InputGroup>
+                      );
+                    }
+                  })}
+                  <InputGroup>
+                    <TextInput
+                      id="textInputAdd"
+                      value={newClaim}
+                      onChange={(value) => {
+                        setNewClaim(value);
+                      }}
+                      aria-label="input with button"
+                    />
+                    <Button
+                      id={"inputButtonAdd"}
+                      icon={<PlusIcon />}
+                      onClick={() => {
+                        if (newClaim) {
+                          claimsSupported.push(newClaim);
+                          field.value &&
+                            (field.value.claimsSupported =
+                              claimsSupported.join(","));
+                          setNewClaim("");
+                          field.onChange(field.value);
+                        }
+                      }}
+                      variant="control"
+                    />
+                  </InputGroup>
+                </>
+              );
+            }}
+          />
+        </FormGroup>
+
+        <ActionGroup>
+          <Button
+            variant="primary"
+            type="submit"
+            data-testid="openid-configuration-tab-save"
+          >
+            {t("save")}
+          </Button>
+          <Button variant="link" onClick={setupForm}>
```

Comment:
Once the form code is split out you can likely call `form.reset()` here and it will go back to the default value.

Reply:
Plain reset() clears all values probably we also need to specify the default values, keeping the setupForm for now.